recipes/ohnetgenerated/all/conanfile.py

```diff
@@ -0,0 +1,125 @@
+from conan import ConanFile
+from conan.tools.apple import fix_apple_shared_install_name, is_apple_os
+from conan.tools.files import get, chdir, copy, mkdir, export_conandata_patches, apply_conandata_patches
+from conan.tools.gnu import Autotools, AutotoolsToolchain
+from conan.tools.layout import basic_layout
+from conan.tools.microsoft import is_msvc, msvc_runtime_flag, NMakeToolchain
+from conan.errors import ConanInvalidConfiguration
+import os
+
+required_conan_version = ">=1.53.0"
+
+
+class OhNetGeneratedConan(ConanFile):
+    name = "ohnetgenerated"
+    description = "Generated proxies and providers for ohMedia and UPnP:AV services"
+    license = "MIT"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/openhome/ohNetGenerated"
+    topics = ("openhome", "ohnet", "ohnetgenerated", "upnp")
+    package_type = "library"
+    settings = "os", "arch", "compiler", "build_type"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+    }
+
+    def _get_openhome_architecture(self, args):
+        if is_apple_os(self):
+            if str(self.settings.arch).startswith("armv8"):
+                openhome_architecture = "arm64"
+                args.extend([f"openhome_architecture={openhome_architecture}", f"detected_openhome_architecture={openhome_architecture}"])
+        return args
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            raise ConanInvalidConfiguration(f"{self.ref} doesn't support shared builds")
+
+    def layout(self):
+        basic_layout(self, src_folder="src")
+
+    def requirements(self):
+        self.requires("ohnet/[>=1.36.5182]", transitive_headers=True, transitive_libs=True)
+
+    def source(self):
+        get(self, **self.conan_data["sources"][self.version], strip_root=True)
+
+    def export_sources(self):
+        export_conandata_patches(self)
+
+    def generate(self):
+        if is_msvc(self):
+            tc = NMakeToolchain(self)
+            tc.generate()
+        else:
+            tc = AutotoolsToolchain(self)
+            tc.make_args.append("-j1")
+            tc.generate()
+
+    def build(self):
+        apply_conandata_patches(self)
+        targets = "proxies devices"
+
+        include_dir = self.dependencies["ohnet"].cpp_info.includedirs[0]
+        lib_dir = self.dependencies["ohnet"].cpp_info.libdirs[0]
```

Comment:
In the other PR you've extracted this into a function, I think it looks better like that, could you also do the same here? Thanks!

Reply:
Sure!